sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkBatchJobsResponse = SparkBatchJobCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJobCollection;
+    };
+};
+
+// @public
+export interface ListSparkSessionsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkSessionsResponse = SparkSessionCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSessionCollection;
+    };
+};
+
+// @public (undocumented)
+export type ListSparkStatementsOptions = OperationOptions;
+
+// @public
+export type ListSparkStatementsResponse = SparkStatementCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatementCollection;
+    };
+};
+
+// @public
+export const logger: import("@azure/logger").AzureLogger;
+
+// @public (undocumented)
+export interface OperationResponse {
+    _response: HttpResponse;
+}
+
+export { PipelineOptions }
+
+// @public
+export type PluginCurrentState = "Preparation" | "ResourceAcquisition" | "Queued" | "Submission" | "Monitoring" | "Cleanup" | "Ended" | string;
+
+// @public (undocumented)
+export type ResetSparkSessionTimeoutOptions = OperationOptions;
+
+// @public
+export type SchedulerCurrentState = "Queued" | "Scheduled" | "Ended" | string;
+
+// @public (undocumented)
+export interface SparkBatchJob {
+    appId?: string;
+    appInfo?: {
+        [propertyName: string]: string;
+    };
+    artifactId?: string;
+    errors?: SparkServiceError[];
+    id: number;
+    jobType?: SparkJobType;
+    // (undocumented)
+    livyInfo?: SparkBatchJobState;
+    logLines?: string[];
+    name?: string;
+    plugin?: SparkServicePlugin;
+    result?: SparkBatchJobResultType;
+    scheduler?: SparkScheduler;
+    sparkPoolName?: string;
+    state?: string;
+    submitterId?: string;
+    submitterName?: string;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    workspaceName?: string;
+}
+
+// @public
+export interface SparkBatchJobCollection {
+    from: number;
+    sessions?: SparkBatchJob[];
+    total: number;
+}
+
+// @public (undocumented)
+export interface SparkBatchJobOptions {
+    // (undocumented)
+    archives?: string[];
+    // (undocumented)
+    arguments?: string[];
+    // (undocumented)
+    artifactId?: string;
+    // (undocumented)
+    className?: string;
+    configuration?: {
+        [propertyName: string]: string;
+    };
+    // (undocumented)
+    driverCores?: number;
+    // (undocumented)
+    driverMemory?: string;
+    // (undocumented)
+    executorCores?: number;
+    // (undocumented)
+    executorCount?: number;
+    // (undocumented)
+    executorMemory?: string;
+    // (undocumented)
+    file: string;
```

Comment:
why is there both `file` and `files`? Isn't `file` the same as a `files` array with one item?

Reply:
[https://livy.apache.org/docs/latest/rest-api.html](https://livy.apache.org/docs/latest/rest-api.html)

That follows the design of Livy. 

---
In reply to: [521609490](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521609490) [](ancestors = 521609490)

Reply:
OK, so you want to stick close to the design of Livy for compatibility and increased comprehension by experts in Livy. Is that correct?

Reply:
yes. we don't want to bring more gaps betweent synapse spark and livy.

---
In reply to: [524825051](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r524825051) [](ancestors = 524825051)